node/src/main/scala/coop/rchain/node/NodeRuntime.scala

```diff
@@ -881,20 +896,17 @@ object NodeRuntime {
           lastFinalizedHeightConstraintChecker,
           reportingCasper
         )
-      casperLoop = for {
-        engine <- engineCell.read
-        _      <- engine.withCasper(_.fetchDependencies, Applicative[F].unit)
-        _ <- Running.maintainRequestedBlocks[F](conf.casper.requestedBlocksTimeout)(
-              Monad[F],
-              rpConfAsk,
-              requestedBlocks,
-              TransportLayer[F],
-              Log[F],
-              Time[F],
-              Metrics[F]
-            )
-        _ <- Time[F].sleep(conf.casper.casperLoopInterval)
-      } yield ()
+      casperLoop = {
+        implicit val br = blockRetriever
+        for {
+          _      <- Time[F].sleep(conf.casper.casperLoopInterval)
```

Comment:
I'd expect for delay to be at the end so that the first execution happens immediately.

Reply:
We don't need it to be executed immediately - as on startup node requests fork-choice tip, and we have logic in CaperLaunch.

But makes sense, maybe we can merge this logic.

Reply:
Order changed